src/Request/ReportRequest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+declare(strict_types=1);
+
+namespace Paytrail\SDK\Request;
+
+use Paytrail\SDK\Exception\ValidationException;
+use Paytrail\SDK\Util\ObjectPropertyConverter;
+
+
+class ReportRequest implements \JsonSerializable
+{
+    const PAYMENT_STATUSES = [
+        'default',
+        'paid',
+        'all',
+        'settled'
+    ];
+
+    private $requestType;
+    private $callbackUrl;
+    private $paymentStatus;
+    private $startDate;
+    private $endDate;
+    private $limit;
+    private $reportFields;
+    private $subMerchant;
+
+    use ObjectPropertyConverter;
+
+    public function validate()
+    {
+        $props = get_object_vars($this);
+
+        if (empty($props['requestType'])) {
+            throw new ValidationException('RequestType can not be empty');
+        }
+
+        if (!in_array($props['requestType'], ['json', 'csv'])) {
+            throw new ValidationException('RequestType must be either "json" or "csv"');
+        }
+
+        if (empty($props['callbackUrl'])) {
+            throw new ValidationException('CallbackUrl can not be empty');
+        }
+
+        if ($props['paymentStatus'] && !in_array($props['paymentStatus'], self::PAYMENT_STATUSES)) {
+            throw new ValidationException('Invalid paymentStatus value');
+        }
+
+        if ($props['limit'] > 50000) {
+            throw new ValidationException('Limit exceeds maximum value of 50000');
+        }
+
+        if ($props['reportFields'] && is_array($props['reportFields'])) {
```

Comment:
I think this should be
```suggestion
        if ($props['reportFields'] && !is_array($props['reportFields'])) {
```

Reply:
Fixed in #44